src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -21,14 +21,16 @@ internal static partial class Crypto
         internal static extern IntPtr ErrReasonErrorString(ulong error);
 
         [DllImport(Libraries.CryptoNative, CharSet = CharSet.Ansi)]
-        private static extern void ErrErrorStringN(ulong e, [Out] StringBuilder buf, int len);
+        private static unsafe extern void ErrErrorStringN(ulong e, byte* buf, int len);
```

Comment:
ErrErrorStringN doesn't return how many bytes were written, and Marshal.PtrToStringAnsi by default walks until a '\0'.  If I were to instead use the byte[] (which I did initially), I'd need to instead use Encoding.UTF8.GetString, which decodes the number of bytes specified (or the whole length of the array), rather than looking for a '\0', e.g. this prints 5:
```C#
byte[] bytes = new byte[] { (byte)'a', (byte)'b', (byte)'\0', (byte)'c', (byte)'d' };
Console.WriteLine(Encoding.UTF8.GetString(bytes).Length);
```
so I'd need to first walk looking for the '\0' and then pass that length to GetString for it to do a second pass.